command/build.go

```diff
@@ -171,6 +178,25 @@ func (c *BuildCommand) RunContext(buildCtx context.Context, cla *BuildArgs) int
 		c.Ui.Say("Debug mode enabled. Builds will not be parallelized.")
 	}
 
+	// TODO find an option that is not managed by a globally shared Publisher.
+	// This build currently enforces a 1:1 mapping that one publisher can be assigned to a single packer config file.
+	// It also requires that each config type implements this ConfiguredArtifactMetadataPublisher to return a configured bucket.
+	ArtifactMetadataPublisher, diags := packerStarter.ConfiguredArtifactMetadataPublisher()
+	if diags.HasErrors() {
+		return writeDiags(c.Ui, nil, diags)
+	}
+
+	if err := ArtifactMetadataPublisher.Initialize(buildCtx); err != nil {
+		diags := hcl.Diagnostics{
+			&hcl.Diagnostic{
+				Summary:  "HCP Packer Registry initialization failed",
+				Detail:   fmt.Sprintf("Unable to open connection to %q at %s\n %s", ArtifactMetadataPublisher.Slug, ArtifactMetadataPublisher.Destination, err),
```

Comment:
`connection` sounds something went wrong with the literal connection between packer and par
```suggestion
				Detail:   fmt.Sprintf("Unable to initialize %q at %s\n %s", ArtifactMetadataPublisher.Slug, ArtifactMetadataPublisher.Destination, err),
```